src/GitHubUtil.js

```diff
@@ -322,6 +322,23 @@ async function updateReference(ref, sha, force) {
     return (await rateLimitedPromise(promise)).object.sha;
 }
 
+async function updateReference(ref, sha, force) {
+    const max = 16 * 1000 + 1; // ~30 sec. overall
+    for (let d = 1000; d < max; d *= 2) {
```

Comment:
Let's simplify and clarify, something along these lines:

```suggestion
    const max = 16 * 1000; // limits total wait to ~30 seconds
    for (let d = 1000; d <= max; d *= 2) {
```